### PR TITLE
support python 3.10 with ray >= 2.0

### LINKED ISSUE #104: Python 3.10 support
# Expectation / Proposal

Python 3.10 is supported by Ray since v2.0.0.
It would be nice to also support it in prefect-ray.

# Traceback / Example

- [X] I would like to [help contribute](https://PrefectHQ.github.io/prefect-ray/#contributing) a pull request to resolve this!

### PR DESCRIPTION
**Pull request by j-tr**
Closes #104

### Example

Ray supports Python 3.10 from version 2.0.0.

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-ray/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-ray/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by j-tr** [comment]
@desertaxle please let me know if you think further changes are necessary in order to support Python 3.10.

File: setup.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
         "Programming Language :: Python :: 3.7",
```
Comment (desertaxle) [inline setup.py:36]: `prefect` doesn't support 3.7 so we can drop this line as well.
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security
 
+## 0.2.6
+
+Released on November 27th, 2023.
+
+### Added
+
+- Support Python 3.10
```
Comment (desertaxle) [inline CHANGELOG.md:28]:
```suggestion
- Support for Python 3.10 - [#105](https://github.com/PrefectHQ/prefect-ray/pull/105)
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
Thanks for the PR @j-tr! We also would want to add tests for Python 3.10 in our CI process. That should be as simple as commenting out this line: https://github.com/PrefectHQ/prefect-ray/blob/340d5d50c4facc3ede0cdd7eab635e6ce34c8ca1/.github/workflows/tests.yml#L16.

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security
 
+## 0.2.6
+
+Released on November 27th, 2023.
```
Comment (desertaxle) [inline CHANGELOG.md:24]:
```suggestion
Released on November 29th, 2023.
```

**Review by desertaxle — APPROVED** [review]

**Pull request merged**